### PR TITLE
Add JSON results download capability

### LINKED ISSUE #234: Add JSON format test results download option to GUI
May also add this as an alternative command line output as opposed to JUnit XML. This wouldn't be intended for parsing by a CI system, but may be useful for other reporting purposes.

### PR DESCRIPTION
**Pull request by andrewbonney**
Resolves #234 

I'm sure the contents of the JSON will need adding to in the future, but this is a starting point.

### CODE REVIEW

File: nmos-test.py [hunk 1]
```diff
@@ -422,54 +423,78 @@ def init_spec_cache():
...
-    return exit_code
-
-
-def print_test_results(results, args):
+def format_test_results(results, format):
+    formatted = None
+    if format == "json":
+        formatted = {"suite": results["def"]["name"],
+                     "url": results["base_url"],
+                     "timestamp": time.time(),
```
Comment (garethsb-sony) [inline nmos-test.py]: Do you want to end up with a timestamp like integer `1558365105.7949972`, or would something like `datetime.datetime.now().strftime("%H:%M:%S.%f")[:-3]` (cf. `Test._current_time`) be better?

File: nmos-test.py [hunk 2]
```diff
@@ -422,54 +423,78 @@ def init_spec_cache():
...
-        TestSuite.to_file(f, [ts], prettyprint=True)
-        print(" * Test results written to file: {}".format(args.output))
-    return exit_code
-
-
-def print_test_results(results, args):
+def format_test_results(results, format):
+    formatted = None
+    if format == "json":
+        formatted = {"suite": results["def"]["name"],
```
Comment (garethsb-sony) [inline nmos-test.py]: Would the test suite id (as required for the `suite` sub-command, or returned from `--list-suites`) be useful instead/as well?

(Though a similar remark could be applied to the JUnit output which uses `results["def"]["class"].__name__`.)

File: nmos-test.py [hunk 3]
```diff
@@ -422,54 +423,78 @@ def init_spec_cache():
...
+                     "url": results["base_url"],
```
Comment (garethsb-sony) [inline nmos-test.py]: This probably applies to simple text output, JUnit output and interactive output as well, but should we be including both (multiple) URLs where the test suite involves more than one API?

File: nmos-test.py (hunk 1 above)
Comment (andrewbonney) [reply nmos-test.py]: I was copying practice of some other testing tools at the time so this was the intention, but I don't have a particular preference. The integer form may be easier to parse however.

File: nmos-test.py (hunk 3 above)
Comment (andrewbonney) [reply nmos-test.py]: Yes, that's been a long standing issue, but probably needs adding as one explicitly.

Comment (garethsb-sony) [reply nmos-test.py]: OK

File: nmos-test.py (hunk 1 above)
Comment (garethsb-sony) [reply nmos-test.py]: OK

File: nmos-test.py (hunk 2 above)
Comment (andrewbonney) [reply nmos-test.py]: Agreed, I've modified this in all cases.

**Pull request merged**